### ISSUE TITLE
Zero-sized icon when using API in Shadow DOM

### ISSUE THREAD

**Issue by heruan** [FortAwesome/Font-Awesome#15449]
**Describe the bug**
When using the SVG core `icon` function to produce an icon inside the Shadow DOM of a custom element, the icon has no size.

**To Reproduce**
This custom element displays no icon, but looking at the DOM you can see the zero-sized `svg`:

```js
import { LitElement, html, css } from 'lit-element';
import { icon } from '@fortawesome/fontawesome-svg-core';
import { faCamera } from '@fortawesome/free-solid-svg-icons';

export class MyElement extends LitElement {
  
  static get styles() {
    return [
      css`
        :host {
          display: flex;
        }
      `
    ];
  }
  
  render() {
    return html`
      ${icon(faCamera).node}
    `;
  }
}

customElements.define('my-element', MyElement);
```
Glitch: https://glitch.com/edit/#!/neon-wisteria

**Expected behavior**
I guess this happens because Shadow DOM isn't affected by external CSS files, such as `fontawesome-svg-core/styles.css`; using external files is also discouraged by [LitElement guidelines](https://lit-element.polymer-project.org/guide/styles#external-stylesheet).

A correct approach would be providing a CSS module to be imported and included in the custom elements styles, e.g.

```js
import { icon, styles } from '@fortawesome/fontawesome-svg-core';
```

and then

```js
  static get styles() {
    return [
      styles, // <- imported from fontawesome-svg-core
      css`
        :host {
          display: flex;
        }
      `
    ];
  }
```

**Comment by tagliala** [comment]
Hi!

Thanks for being part of the Font Awesome Community and thanks for the detailed report.

Does https://github.com/FortAwesome/Font-Awesome/issues/5274#issuecomment-265872328 solve this issue?

**Comment by heruan** [comment]
Thank you for the pointer! That is the workaround I'm using, but importing external files in Shadow DOM is discouraged in favor of modularized definitions.

Since FA provides this great JavaScript API compatible with ES modules, it would be great to have support also for modularized styles!

**Comment by heruan** [comment]
A better workaround is to create a `CSSResult` element such as:

```js
import { css } from 'lit-element';

const faStyles = css`
svg:not(:root).svg-inline--fa {
  overflow: visible; }

/* ... the rest of fontawesome-svg-core/styles.css */

.fad.fa-inverse {
  color: #fff; }
`;

export { faStyles };
```

and import it where necessary.

**Comment by tagliala** [comment]
you're welcome

> ```js
> import { icon, styles } from '@fortawesome/fontawesome-svg-core';
> ```

So we just need to export `styles` and that will provide a better support for this feature?

**Comment by heruan** [comment]
> So we just need to export `styles` and that will provide a better support for this feature?

Yep, but it should be a `CSSResult` instance and you might not want to depend on `lit-element`. Or, you can just export the styles as a string and one could use `unsafeCSS` to import them, e.g.

```js
import { LitElement, html, css, unsafeCSS } from 'lit-element';
import { icon, styles } from '@fortawesome/fontawesome-svg-core';

export class MyElement extends LitElement {

  static get styles() {
    return [
      unsafeCSS(styles),
      css` /* my styles */ `
    ];
  }
  // ... rest of class
}
```
